deployer/hub.py

```diff
@@ -140,6 +142,46 @@ def auth_kubeconfig(self):
             os.environ['KUBECONFIG'] = decrypted_key_path
             yield
 
+    def auth_aws(self):
+        config = self.spec['aws']
+        key_path = config['key']
+        cluster_type = config['clusterType']
+        cluster_name = config['clusterName']
+        region = config['region']
+
+        with tempfile.NamedTemporaryFile() as kubeconfig:
+            orig_kubeconfig = os.environ.get('KUBECONFIG', None)
+            orig_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID', None)
+            orig_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY', None)
+            try:
+                os.environ['KUBECONFIG'] = kubeconfig.name
+                with decrypt_file(key_path) as decrypted_key_path:
+
+                    decrypted_key_abspath = os.path.abspath(decrypted_key_path)
+                    if not os.path.isfile(decrypted_key_abspath):
+                        raise FileNotFoundError(
+                            'The decrypted key file does not exist')
+                    with open(decrypted_key_abspath) as f:
+                        creds = json.load(f)
+
+                    os.environ['AWS_ACCESS_KEY_ID'] = creds["AccessKey"]['AccessKeyId']
+                    os.environ['AWS_SECRET_ACCESS_KEY'] = creds["AccessKey"]['SecretAccessKey']
+
+                subprocess.check_call([
```

Comment:
Do we want the KUBECONFIG to be set as late as possible?

I believe the sops encryption operation can take a while. Will there be race conditions in the logic using this function? I'm a bit vague on the use of generators. That is a separate question in a way that also may relate to the question on where to set KUBECONFIG.

It vaguely feels better that its set just before calling aws eks update-kubeconfig to minimize the time its influencing things and such.

Reply:
File operations here should be blocking, am I right? So, the subprocess call should proceed after we get the kubeconfig file properly written (and closed) and the AWS env variables properly set up.
Btw, I have not seen that race condition so far (it does not mean it could not happen if I am wrong about the above :wink:).

Reply:
> It vaguely feels better that its set just before calling aws eks update-kubeconfig to minimize the time its influencing things and such.

I was writing a reply to this one and get posted before it was ready :wink:
In any case, I now see where you are aiming at... I feel better as well if we set the KUBECONFIG env variable just before the `update-config` command, I will make the change.